src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const stateHistory = [];
+
+  for (const obj of actions) {
+    if (obj.type === 'addProperties') {
+      Object.assign(stateClone, obj.extraData);
+      stateHistory.push({ ...stateClone });
```

Comment:
Instead of pushing the clone to `stateHistory` inside each `if` block, you could do it once at the end of the loop to follow the DRY (Don't Repeat Yourself) principle.